### ISSUE TITLE
[Feature Request] BGO / Decor settings? (Like SMBX2)

### ISSUE THREAD

**Issue by LoveBodhi** [Wohlstand/TheXTech#114]
Priority is same as Z-order.

Animated BGOs are very limited.

There's also a BGO Cutoff (Only for custom side), which is a vanilla bug
[Test 10.zip](https://github.com/Wohlstand/TheXTech/files/6543599/Test.10.zip)
![Scr_2021-05-26_11-03-38](https://user-images.githubusercontent.com/37089530/119596847-867d9d00-be12-11eb-98fb-22994a8cc195.png)

**Comment by P-Star7** [comment]
My suggestions:

Auto-resize based on height divided by frames
.txt codes (all except frames and zdepth are defaultly 0/negative)

frames
zdepth (or some other synonym)
water (makes the splash)
climbable (fence)
bumper (SMB3 and SMW platform bumpers)
keyhole
hoopsterstree (which the hoopster reverses y-speed when touches this)
path (0/default = not a path, 1 = horizontal, 2 = vertical, 3 = right-up, 4 = right-down)
door (the non-zero number which is put here is the effect that shows when enter a door warp on top of it)
door2 (effect number, see below)
dooreffectanimation (0 = door's effect plays and reverses when the player exits on other side and door2 definition is unnecessary for this door, 1 = door effect is shown when enter the door, door2 effect is shown when exit the door)
dooroffsetx (affects the position of the effect for differently sized doors like SMB2 doors, SMW ghost house door)
dooroffsety

**Comment by LoveBodhi** [comment]
Plus Circular, 1x2 diagonal, 2x1 diagonal, etc. (like SMBX2 did)

**Comment by P-Star7** [comment]
> Plus Circular, 1x2 diagonal, 2x1 diagonal, etc. (like SMBX2 did)
> 
> Also "framespeed"

You are right about framespeed, I forgot, but it only should be SMBX 1.3's settings as the first focus.

**Comment by LoveBodhi** [comment]
`splashoffset`: Y-position of the splash effect.

**Comment by LoveBodhi** [comment]
Should this be merged into #427 ?
